Nonexisting named map tokens aren't handled properly

```
[2015-03-03 15:19:12.795] [ERROR] [default] - Uncaught exception: Error: Invalid or nonexistent map configuration token 'f5c30e9593027f8dbca79ad018f489cc'
    at Function.parseRecord (/home/ubuntu/www/node-windshaft/shared/node_modules/windshaft/lib/windshaft/storages/mapstore.js:117:15)
    at next (/home/ubuntu/www/node-windshaft/shared/node_modules/step/lib/step.js:51:23)
    at Function.releaseRedisClient (/home/ubuntu/www/node-windshaft/shared/node_modules/windshaft/lib/windshaft/storages/mapstore.js:74:29)
    at next (/home/ubuntu/www/node-windshaft/shared/node_modules/step/lib/step.js:51:23)
    at try_callback (/home/ubuntu/www/node-windshaft/shared/node_modules/redis-mpool/node_modules/redis/index.js:592:9)
    at RedisClient.return_reply (/home/ubuntu/www/node-windshaft/shared/node_modules/redis-mpool/node_modules/redis/index.js:685:13)
    at HiredisReplyParser.<anonymous> (/home/ubuntu/www/node-windshaft/shared/node_modules/redis-mpool/node_modules/redis/index.js:321:14)
    at HiredisReplyParser.EventEmitter.emit (events.js:95:17)
    at HiredisReplyParser.execute (/home/ubuntu/www/node-windshaft/shared/node_modules/redis-mpool/node_modules/redis/lib/parser/hiredis.js:43:18)
    at RedisClient.on_data (/home/ubuntu/www/node-windshaft/shared/node_modules/redis-mpool/node_modules/redis/index.js:547:27)
```

and the request hangs

--- a/lib/cartodb/server_options.js
+++ b/lib/cartodb/server_options.js
@@ -400,7 +400,9 @@ module.exports = function(redisPool) {
         }
 
         mapStore.load(layergroup_id, function(err, mapConfig) {
-            assert.ifError(err);
+            if (err) {
+                return callback(err);
+            }
 
             var authorized = me.templateMaps.isAuthorized(mapConfig.obj().template, auth_token);
 
@@ -475,7 +477,9 @@ module.exports = function(redisPool) {
                 });
             },
             function checkSignAuthorized(err, signed_by){
-                assert.ifError(err);
+                if (err) {
+                    return callback(err);
+                }
 
                 if ( ! signed_by ) {
                   // request not authorized by signer.

--- a/test/acceptance/templates.js
+++ b/test/acceptance/templates.js
@@ -2045,6 +2045,68 @@ describe('template_api', function() {
         );
     });
 
+    describe('named map nonexistent tokens', function() {
+        var username = 'localhost';
+        var templateHash = 'deadbeef';
+        var nonexistentToken = 'wadus';
+
+        function request(token) {
+            return {
+                url: '/api/v1/map/' + token + '/all/0/0/0.png',
+                method: 'GET',
+                headers: {
+                    host: username
+                },
+                encoding: 'binary'
+            };
+        }
+
+        var expectedResponse = {
+            headers: {
+                'Content-Type': 'application/json; charset=utf-8'
+            },
+            status: 400
+        };
+
+        function checkTileFn(done) {
+            return function checkTile(res, err) {
+                if (err) {
+                    return done(err);
+                }
+                assert.deepEqual(JSON.parse(res.body), {
+                    error: "Invalid or nonexistent map configuration token '" + nonexistentToken + "'"
+                });
+
+                done();
+            };
+        }
+
+        it("returns an error for named map nonexistent tokens", function(done) {
+
+            var nonexistentNamedMapToken = username + '@' + templateHash + '@' + nonexistentToken;
+
+            assert.response(
+                server,
+                request(nonexistentNamedMapToken),
+                expectedResponse,
+                checkTileFn(done)
+            );
+        });
+
+        it("returns an error for named map nonexistent tokens without template hash", function(done) {
+
+            var nonexistentNamedMapToken = username + '@' + nonexistentToken;
+
+            assert.response(
+                server,
+                request(nonexistentNamedMapToken),
+                expectedResponse,
+                checkTileFn(done)
+            );
+        });
+
+    });
+
     after(function(done) {
 
         // This test will add map_style records, like

